azerothcore/azerothcore-wotlk
# feat(Core/QuestTracker): implement quest tracker manager

## CHANGES PROPOSED:
- Implement queue 
- Add separate class
- Rework configuration for system
```ini
#
#    QuestTracker.Enable
#        Description: Store data in the database about quest completion and abandonment to help finding bugged quests.
#        Default:     0  - (Disabled)
#                     1  - (Enabled)
#
#    QuestTracker.Queue.Enable
#        Description: Enable queue
#        Default:     0  - (Disabled)
#                     1  - (Enabled)
#
#    QuestTracker.Queue.Delay
#        Description: Delay for update queue in seconds. 1 seconds minimun
#        Default:     10
#

QuestTracker.Enable = 0
QuestTracker.Queue.Enable = 0
QuestTracker.Queue.Delay = 10
```

## ISSUES ADDRESSED:
- #4005

## Tests Performed:
- https://gist.github.com/Winfidonarleyan/e952b58c701f272ac62faa303253b9bd

## How to Test the Changes:
- Download this module https://github.com/WarheadCore/mod-quest-tracker-test-ac
- Compile core with this PR
- Add macro in-game
```
.questt 500 0
.questt 500 1
.questt 500 2
.questt 500 3
```
- Check worldserver console
- Check DB table `quest_tracker`

## Known Issues and TODO List:

## Target Branch(es):






 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
